internal/util/debug/debug.go

```diff
@@ -118,73 +141,57 @@ func Listen(opts *ListenOpts) (*Handler, error) {
 
 	http.HandleFunc("/debug/archive", func(rw http.ResponseWriter, req *http.Request) {
 		rw.Header().Set("Content-Type", "application/zip")
-
 		rw.Header().Set(
 			"Content-Disposition",
-			fmt.Sprintf("attachment; filename=FerretDB-debug-%d.zip", time.Now().UnixMilli()),
+			fmt.Sprintf("attachment; filename=ferretdb-%s.zip", time.Now().Format("2006-01-02-15-04-05")),
 		)
 
 		ctx := req.Context()
 		zipWriter := zip.NewWriter(rw)
+		errs := map[string]error{}
 
 		defer func() {
-			if err := zipWriter.Close(); err != nil {
-				l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
-				return
-			}
-		}()
-
-		metricsFile, err := zipWriter.Create("metrics")
-		if err != nil {
-			l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
-			return
-		}
-
-		debugAddr := ctx.Value(http.LocalAddrContextKey).(net.Addr)
-
-		// we use *http.Request instead of http.Get function to provide the ctx
-		scrapeReq := must.NotFail(http.NewRequestWithContext(
-			ctx,
-			http.MethodGet,
-			fmt.Sprintf("http://%s%s", debugAddr.String(), "/debug/metrics"),
-			nil,
-		))
-
-		resp, err := http.DefaultClient.Do(scrapeReq)
-		if err != nil {
-			l.ErrorContext(ctx, "Archive handler failed - metrics failed", logging.Error(err))
-			return
-		}
-
-		_, err = io.Copy(metricsFile, resp.Body)
-
-		_ = resp.Body.Close()
-
-		if err != nil {
-			l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
-			return
-		}
+			files := maps.Keys(errs)
+			slices.Sort(files)
 
-		heapFile, err := zipWriter.Create("heap")
-		if err != nil {
-			l.ErrorContext(ctx, "Archive handler failed", logging.Error(err))
-			return
-		}
+			var b bytes.Buffer
+			for _, f := range files {
+				b.WriteString(fmt.Sprintf("%s: %v\n", f, errs[f]))
+			}
 
-		scrapeReq.URL.Path = "/debug/pprof/heap"
+			if err := addToZip(zipWriter, "errors.txt", io.NopCloser(&b)); err != nil {
```

Comment:
Do we always want to include `errors.txt` even if all files were successfully written?

Reply:
Yeah, it is not that important and makes code simpler